### PR TITLE
Pods 2449 - Working knowledge section

### PR DESCRIPTION
**Pull request by madhuranakate**

### CODE REVIEW

File: test/utils/navigators/AdviserNavigatorSpec.scala [hunk 1]
```diff
@@ -46,33 +47,36 @@ class AdviserNavigatorSpec extends SpecBase with NavigatorBehaviour {
...
-    (CheckYourAnswersId, emptyAnswers, haveMoreChangesPage, false, None, false)
+    (CheckYourAnswersId, emptyAnswers, haveMoreChangesPage, false, None, false),
+    (CheckYourAnswersId, declarationPensionAdvisorTrue, variationDeclarationFitAndProperPage, false, None, false)
   )
 
   navigator.getClass.getSimpleName must {
     appRunning()
-    behave like nonMatchingNavigator(navigator)
-    behave like navigatorWithRoutes(navigator, FakeUserAnswersCacheConnector, routes(), dataDescriber)
+    //behave like nonMatchingNavigator(navigator)
```
Comment (abhijeetardale) [inline test/utils/navigators/AdviserNavigatorSpec.scala]: please remove  commented code

File: app/controllers/register/VariationWorkingKnowledgeController.scala [hunk 2]
```diff
@@ -67,16 +66,26 @@ class VariationWorkingKnowledgeController @Inject()(
...
+          val resultOfSaveDeclarationFlag = mode match {
+            case CheckUpdateMode =>
+              cacheConnector.save (request.externalId, PAInDeclarationJourneyId, true)
+            case _ =>
+              Future.successful(())
+          }
+
+          resultOfSaveDeclarationFlag.flatMap(_ =>
+            cacheConnector.save(request.externalId, IsNewAdviserId, !value).flatMap(_ =>
+              if (hasAnswerChanged) {
```
Comment (abhijeetardale) [inline app/controllers/register/VariationWorkingKnowledgeController.scala]: can we removed hasAnswerChanged conditions? as it was decided we are not going to compare data

File: test/utils/navigators/AdviserNavigatorSpec.scala (hunk 1 above)
Comment (grantrobson) [reply test/utils/navigators/AdviserNavigatorSpec.scala]: done

File: app/controllers/register/VariationWorkingKnowledgeController.scala (hunk 2 above)
Comment (grantrobson) [reply app/controllers/register/VariationWorkingKnowledgeController.scala]: done

**Pull request merged**